server/src/main/java/io/crate/expression/scalar/arithmetic/TimestampArithmeticScalar.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.arithmetic;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.joda.time.Period;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public class TimestampArithmeticScalar extends Scalar<Period, Object> implements BiFunction<Long, Long, Period> {
+
+    public static void register(ScalarFunctionModule module) {
+        for (var timestampType : List.of(DataTypes.TIMESTAMP, DataTypes.TIMESTAMPZ)) {
+            module.register(
+                Signature.scalar(
+                    ArithmeticFunctions.Names.SUBTRACT,
+                    timestampType.getTypeSignature(),
+                    timestampType.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature()
+                ),
```

Comment:
do we need `.withForbiddenCoersion()` here?

Reply:
Should be okay to leave it out